/*
 * Copyright (c) 2024 Greg Padiasek
 * Distributed under the terms of the MIT License.
 * See accompanying file LICENSE or copy at http://opensource.org/licenses/MIT
 */

//
// SEEGNIFY graph messages
//

syntax = "proto2";
package seegnify.graph;

// GetWeights request

message GetWeights {
}

// GetWeights response

message GetWeightsResponse {
  required bytes weights = 1;
}

// SetWeights request

message SetWeights {
  required bytes weights = 1;
}

// Update Weights request

message UpdWeights {
  required bytes update = 1;
}

// Success response

message SuccessResponse {
}

// Failure response

message ErrorResponse {
  required uint32 status = 1;  
  required string message = 2;
}

// Message envelopes (union)

message Request {
  oneof request {
    seegnify.graph.GetWeights get_weights = 10;
    seegnify.graph.SetWeights set_weights = 11;
    seegnify.graph.UpdWeights upd_weights = 12;
  }
}

message Response {
  oneof response {
    seegnify.graph.GetWeightsResponse get_weights = 11;
    seegnify.graph.SuccessResponse success = 12;
    seegnify.graph.ErrorResponse error = 13;
  }
}

